### DISCUSSION TITLE
Cannot push

### DISCUSSION THREAD

**Discussion by belbernard** [community/community#122833] [Codespaces]
### Select Topic Area

Bug

### Body

Hello,

I successfully uploaded changes to the repository:
https://github.com/bolprocessor/bolprocessor/tree/graphics-for-BP3

I am the admin of two more repositories:
https://github.com/bolprocessor/php-frontend
https://github.com/bolprocessor/bp3-ctests/

However, I failed uploading changes to these ones. I am using the same ssh key and the same paraphrase. I am getting error messages in the style:

git push
Enter passphrase for key '/Users/myname/.ssh/id_rsa': 
ERROR: Permission to bolprocessor/php-frontend.git denied to deploy key
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists.

**Comment by ghost** [comment]
Check SSH Key Configuration , duble ckeck the key and the permissions for "php-frontend" and "bp3-ctests". Verify that you're entering the correct passphrase for your SSH key when prompted. If you're unsure, you might consider regenerating the SSH key pair

**Reply by belbernard** [reply]
Thanks for the suggestions! I will try regenerating the SSH key pair. However, it is strange that the same key and paraphrase work for the first repository. Does it mean that I need to create a new key for each repository?

**Reply by ghost** [reply]
As you know you can use the same SSH key pair for multiple repositories. However, there are different methods for managing access to repositories, and sometimes the setup may require different configurations

**Reply by belbernard** [reply]
This is strange because I am the owner of the 3 repositories and the SSH key + paraphrase worked perfectly for all of them. Now the error "Permission to bolprocessor/php-frontend.git denied to deploy key" suggests that I should do something on GitHub's end to validate the key for this particular repository.

**Reply by belbernard** [reply]
The .git/config file of "php-frontend" (which doesn't work) is:

[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
	ignorecase = true
	precomposeunicode = true
[remote "destination"]
	url = git@github.com:bolprocessor/php-frontend
	fetch = +refs/heads/*:refs/remotes/destination/*
[branch "master"]
	remote = destination
	merge = refs/heads/master

The .git/config file of "graphics-for-BP3" (which works) is:

[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
	ignorecase = true
	precomposeunicode = true
[remote "destination"]
	url = git@github.com:bolprocessor/bolprocessor
	fetch = +refs/heads/*:refs/remotes/destination/*
[branch "master"]
	remote = destination
	merge = refs/heads/master
[branch "BP3-develop"]
	remote = destination
	merge = refs/heads/BP3-develop
[branch "graphics-for-BP3"]
	remote = destination
	merge = refs/heads/graphics-for-BP3
[pull]
	rebase = false
	rebase = false

**Comment by 22coffeeboi** [comment]
It seems like you're encountering a permission issue when trying to push changes to your repositories "bolprocessor/php-frontend" and "bolprocessor/bp3-ctests". The error message indicates that permission to push changes is being denied to the deploy key associated with your SSH key.

Here are a few steps you can take to troubleshoot and resolve this issue:

Check Deploy Key Configuration: Make sure that the SSH key you're using as a deploy key has the appropriate permissions configured in the settings of the repositories "bolprocessor/php-frontend" and "bolprocessor/bp3-ctests". Ensure that the deploy key has both read and write access if you intend to push changes.
Verify SSH Key Setup: Double-check that the SSH key you're using is correctly configured and added to your GitHub account. Ensure that there are no typos or errors in the SSH key or passphrase.
Confirm Repository Existence: Ensure that the repositories "bolprocessor/php-frontend" and "bolprocessor/bp3-ctests" exist on GitHub under your account or the organization's account. The error message suggests that Git is unable to find the remote repository.
Check Access Rights: Ensure that you have the correct access rights to push changes to these repositories. If you're the admin of the repositories, you should have the necessary permissions by default, but it's worth confirming.
Try Cloning the Repository: If you suspect there might be an issue with the repository URL or configuration, try cloning the repository again using the SSH URL to see if you can access it successfully.
Contact GitHub Support: If you've tried all the above steps and still can't resolve the issue, consider reaching out to GitHub support for further assistance. They can provide insights into any potential issues with permissions or repository configurations.
By following these steps, you should be able to diagnose and resolve the permission issue preventing you from pushing changes to your repositories

**Reply by belbernard** [reply]
OK, I am going to check all these points in great detail.
I tried cloning one of these repositories. The "pull" worked perfectly, but when I tried to push it first asks for my username and password, then it reminded me tells me that "Support for password authentication was removed". However, it suggested that I create a SSH key, which I already have…
Can you give me link to reach GitHub support?

**Comment by belbernard** [comment]
I tried this (https://docs.github.com/en/authentication/connecting-to-github-with-ssh/testing-your-ssh-connection):

belbernard@macbookobernard bolprocessor_php % ssh -T git@github.com
Enter passphrase for key '/Users/belbernard/.ssh/id_rsa': 
Hi bolprocessor/bolprocessor! You've successfully authenticated, but GitHub does not provide shell access.

The manual says: "Verify that the resulting message contains your username." But my username is "belbernard", not "bolprocessor/bolprocessor".

**Reply by belbernard** [reply]
The problem is solved by "adding a SSH key" and entering the public key into the settings. Thanks for your support!

**Comment by SYNAX07** [comment]
It looks like you're encountering permission issues when trying to push changes to the repositories `bolprocessor/php-frontend` and `bolprocessor/bp3-ctests` using a deploy key. The error message suggests that the deploy key associated with your SSH key is being denied permission to access these repositories.

Here are some steps you can take to troubleshoot and resolve this issue:

1. **Verify Deploy Key Setup**:
   Ensure that the SSH key you're using (`/Users/myname/.ssh/id_rsa`) is indeed added as a deploy key to the repositories `bolprocessor/php-frontend` and `bolprocessor/bp3-ctests`. Navigate to the repository settings on GitHub and check under "Deploy keys" to confirm that the correct SSH key is added with read/write permissions.

2. **Check SSH Key Permissions**:
   Double-check the permissions and configuration of your SSH key (`id_rsa`). The private key file (`id_rsa`) should have restrictive permissions (`chmod 600 ~/.ssh/id_rsa`) to ensure it's only accessible by your user.

3. **Confirm Passphrase Entry**:
   When prompted for the passphrase during `git push`, make sure you are entering the correct passphrase associated with your SSH key. Ensure there are no typos or accidental changes.

4. **Verify Repository URL**:
   Ensure that the repository URL (`git@github.com:bolprocessor/php-frontend.git` and `git@github.com:bolprocessor/bp3-ctests.git`) is correctly configured in your local Git repository. You can check this using:
   ```bash
   git remote -v
   ```

5. **SSH Configuration**:
   Check if there are any specific SSH configurations (`~/.ssh/config`) that might be affecting your SSH connection to GitHub. Ensure that there are no conflicting configurations or settings.

6. **Test SSH Connection**:
   Use the following command to test your SSH connection to GitHub:
   ```bash
   ssh -T git@github.com
   ```
   This should display a message confirming a successful connection to GitHub.

7. **Re-add Deploy Key**:
   If you suspect issues with the deploy key, try removing and re-adding the SSH key as a deploy key in the repository settings on GitHub. Make sure you copy the public key (`id_rsa.pub`) to the deploy key section.

8. **Check Access Restrictions**:
   Ensure that there are no repository-level access restrictions (e.g., branch protection rules, repository visibility settings) that could be preventing the deploy key from pushing changes.

9. **Review GitHub Logs**:
   Check the repository's access logs on GitHub (`Settings` -> `Audit log`) to see if there are any specific error messages or events related to the failed pushes.

If you continue to encounter issues after trying these steps, consider reaching out to GitHub support for further assistance, especially if there are specific permissions or settings related to the repositories that need to be addressed.
